src/runtime/virtcontainers/clh.go

```diff
@@ -693,6 +690,9 @@ func (clh *cloudHypervisor) StartVM(ctx context.Context, timeout int) error {
 	}
 	clh.state.PID = pid
 
+	ctx, cancel := context.WithTimeout(context.Background(), clh.getClhAPITimeout()*time.Second)
```

Comment:
Is it possible to create the context using the existing one (ctx passed in) as the parent context? I didn't have time to extensively look through the code but a quick search showed that subsequent calls from StartVM and beyond create trace spans from this context, and when a new context from background is used it creates a span with no parent unattached to the root. However, I'm not sure if the parent context[s] also had a timeout and if this conflicts. @jodh-intel 

Despite that, I'm approving since this use of context was already approved and you're only changing the location here.

Reply:
Agree with @cmaf, use the input `ctx` can keep passing information to the next function calls.

Reply:
Got it, fixed.